### PR TITLE
generator files cleanup

### LINKED ISSUE #17: App template should use relative paths for dev dependencies
So instead of:

```
"dev": "supervisor -k -e ...
```

Should be:

```
"dev": "./node_modules/.bin/supervisor -k -e ...
```

For all default npm commands in the app template.

### LINKED ISSUE #63: Some general cleanup needed
- sampleApp folder no longer needed
- dependencies of app template are way out of date

### PR DESCRIPTION
**Pull request by lyomr**
closes #63 and closes #17

### CODE REVIEW

**Comment by alallier** [comment]
@kethinov you wanted the package-json file committed so it will need to be templated too.

**Comment by kethinov** [comment]
package-lock you mean? no that doesn't need to be templating. it is auto generated by a user when they first run `npm i`

**Comment by alallier** [comment]
Yes I meant the lock and sorry I see the lock file was committed for the actual generator that is good, for some reason I though it was the one from the generated sample app.

File: generators/app/templates/templatedPackage.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
   "scripts": {
     "start": "npm run prod",
     "prod": "supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -prod",
-    "dev": "supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -dev",
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: the dependencies of the sample app itself need to be updated too in this file

File: generators/app/templates/templatedPackage.json [hunk 2]
```diff
@@ -79,8 +79,8 @@
   },
   "scripts": {
     "start": "npm run prod",
-    "prod": "supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -prod",
-    "dev": "supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -dev",
-    "test": "jshint app.js"
+    "prod": "--scripts-prepend-node-path supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -prod",
+    "dev": "--scripts-prepend-node-path supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -dev",
+    "test": "--scripts-prepend-node-path jshint app.js"
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: still need to update the deps in this file

File: generators/app/index.js [hunk 3]
```diff
@@ -258,7 +258,7 @@ module.exports = generators.Base.extend({
           this.disableLogger = answers.disableLogger ? answers.disableLogger :  this.options.disableLogger || 'false';
           this.noMinify = answers.noMinify ? answers.noMinify :  this.options.noMinify || 'false';
           this.enableValidator = this.options.enableValidator || 'false';
-          this.htmlValidator = this.options.enableValidator || '{validator: "http://html5.validator.nu", format: "text", suppressWarnings: true}';
+          this.htmlValidator = this.options.enableValidator || '{"validator": "http://html5.validator.nu", "format": "text", "suppressWarnings": true}';
```
Comment (kethinov) [inline generators/app/index.js]: change this to:

```
this.htmlValidator = this.options.enableValidator || '{"format": "text", "suppressWarnings": true}';
```

Comment (kethinov) [inline generators/app/index.js]: hmm that was super out of date...

make sure you audit all the default values to make sure they match the defaults that are in here: https://github.com/rooseveltframework/roosevelt/blob/master/lib/sourceParams.js

Comment (kethinov) [inline generators/app/index.js]: it may be a bit hard to follow but, for example:

```js
port: process.env.HTTP_PORT || process.env.NODE_PORT || params.port || pkg.rooseveltConfig.port || 43711
```

in the above example, the default is the last condition of the `or` chain, so, the default would be 43711.

File: generators/app/index.js [hunk 4]
```diff
@@ -322,13 +327,16 @@ module.exports = generators.Base.extend({
...
           this.htmlMinify = this.options.htmlMinify || '{"override": true, "exception_url": false, "htmlMinifier": {"html5": true}}';
           this.cssPath = answers.cssPath ? answers.cssPath :  this.options.cssPath || 'css';
           this.cssCompiler = answers.cssCompiler ? answers.cssCompiler :  this.options.cssCompiler || '{"nodeModule": "roosevelt-less", "params": {"compress": true}}';
           this.cssCompilerWhitelist = answers.cssCompilerWhitelist ? answers.cssCompilerWhitelist :  this.options.cssCompilerWhitelist || 'null';
           this.cssCompiledOutput = answers.cssCompiledOutput ? answers.cssCompiledOutput :  this.options.cssCompiledOutput || '.build/css';
           this.jsPath = answers.jsPath ? answers.jsPath :  this.options.jsPath || 'js';
+          this.bundledJsPath = answers.bundledJsPath ? answers.bundledJsPath :  this.options.bundledJsPath || '.bundled';
+          this.exposeBundles = answers.exposeBundles ? answers.exposeBundles :  this.options.exposeBundles || 'true';
+          this.browserifyBundles = answers.browserifyBundles ? answers.browserifyBundles :  this.options.browserifyBundles || '[]';
```
Comment (kethinov) [inline generators/app/index.js]: should be `[]` instead of `'[]'`

h/t @Autre31415

Comment (Autre31415) [inline generators/app/index.js]: The htmlMinify param is being installed wrong. Coming up as `"htmlMinify": "htmlMinify"` in a fresh sample app. My guess based on line 331 is that `this.options.htmlMinify` is set to that string and preventing the default from being set.

File: generators/app/index.js [hunk 5]
```diff
@@ -322,13 +327,16 @@ module.exports = generators.Base.extend({
           this.error404 = answers.error404 ? answers.error404 :  this.options.error404 || '404.js';
           this.error5xx = answers.error5xx ? answers.error5xx :  this.options.error5xx || '5xx.js';
           this.error503 = answers.error503 ? answers.error503 :  this.options.error503 || '503.js';
-          this.staticsRoot = answers.staticsRoot ? answers.staticsRoot :  this.options.staticsRoot || 'statics';
+          this.staticsRoot = answers.staticsRoot ? answers.staticsRoot :  this.options.staticsRoot || 'statics/';
```
Comment (kethinov) [inline generators/app/index.js]: change this back to what it was

File: generators/app/index.js (hunk 4 above)
Comment (lyomr) [inline generators/app/index.js]: Fixed in a commit down below https://github.com/rooseveltframework/generator-roosevelt/pull/65/commits/cd342205f92448a606f1a927b0b6d080f070e851
